lib/parser/grammar.h

```diff
@@ -1088,9 +1088,9 @@ TYPE_PARSER(construct<ExplicitCoshapeSpec>(
 // a need for forced look-ahead.
 TYPE_PARSER(
     construct<ArraySpec>(parenthesized(nonemptyList(explicitShapeSpec))) ||
+    construct<ArraySpec>(parenthesized(deferredShapeSpecList)) ||
```

Comment:
A comment here about the need to attempt `deferredShapeSpecList` first to resolve a potential ambiguity with assumed shape specs would be great.

Reply:
OK